File: java/class/Queue.java
public class Queue {
    private int length;
    private int[] queue;
    public Queue(int[] queue) {
        if (queue == null) {
            throw new IllegalArgumentException("queue must not be null");
        }
        this.queue = queue;
        this.length = queue.length;
    }

    public void enqueue(int element) {
        int[] newQueue = new int[length + 1];
        System.arraycopy(queue, 0, newQueue, 0, length);
        newQueue[length] = element;
        queue = newQueue;
        length++;
    }

    public int dequeue() {
        if (length == 0) {
            throw new IllegalStateException("queue is empty");
        }
        int element = queue[0];
        int[] newQueue = new int[length - 1];
        System.arraycopy(queue, 1, newQueue, 0, length - 1);
        queue = newQueue;
        length--;
        return element;
    }

    public int[] getQueue() {
        return queue;
    }

    public int getLength() {
        return length;
    }

    public void clear() {
        queue = new int[0];
        length = 0;
    }

    public String toString() {
        return this.getQueue().toString();
    }
}

File: ts/class/Queue.ts
export class Queue extends Object {
    length: number;
    #queue: any[];
    constructor(queue = []) {
        super();
        if (!Array.isArray(queue)) {
            throw new Error("Input must be an array");
        }
        this.#queue = queue;
        this.length = queue.length;
    }

    enqueue(value: any) {
        this.#queue.push(value);
        this.length++;
    }

    dequeue() {
        if (this.length === 0) {
            throw new Error("Queue is empty");
        }
        this.length--;
        return this.#queue.shift();
    }

    get getQueue() {
        return this.#queue;
    }

    clear() {
        this.#queue = [];
        this.length = 0;
    }

    override toString() {
        return this.#queue.toString();
    }
}